### PR TITLE
Fix SDSS CrossID method not working for spec data and for DR17 (photo and spec)

### PR DESCRIPTION
**Pull request by nmcardoso**
# Sumary

This PR aims to fix the following errors in the `astroquery.sdss.SDSS.query_crossid_async()` method:

1. Fix documentation typo about default value of the `radius` param (bcef71672fdb6d47a535479f7252af71cef34c63)
2. Fix spec query not working for DR >= 12 (777f80b4db026360cd461d93498d7da8e86c8f14)
3. Fix photo and spec query not working for DR 17 (4fb7217d125d42ea52a2b5d47f2cfc82050228cb)

# Details about each fix

## 1. Documentation typo

Changed documentation to the right value of the default radius param

Commit: bcef71672fdb6d47a535479f7252af71cef34c63

## 2. Spec query for DR >= 12

Spec queries for DR >= 12 are resulting in a 500 (Internal Server Error) HTTP response 
becouse the query string formed by this method is invalid when requesting columns of
SpecObj table. The query properly includes the spec columns but forgot the 
`JOIN` statement with the SpecObj table.

Commit: 777f80b4db026360cd461d93498d7da8e86c8f14

## 3. Photo and spec query for DR 17

Both kind of queries to the DR 17 API results in a 500 (Internal Server Error) HTTP response 
becouse this API requires parameters to be sent in the request body as a payload.
Passing the parans via `data` attribute instead of the `params` attribute of the
http client solves the issue, since previous data releases (DR >= 12) allows this approach too.

Commit: 4fb7217d125d42ea52a2b5d47f2cfc82050228cb

# Tests Results

All DR >= 12 where tested for photo and spec query. 
All requests returned 200 status code and the method returned an correct `astropy.table.table.Table` instance.

Test code:

```python
from astroquery.sdss import SDSS
from astropy import coordinates as coord
import numpy as np
from astropy import units as u
from astroquery import log
log.setLevel('TRACE')

pos = coord.SkyCoord(
  ra=np.array([359.970818400315, 0.1286587799259]), 
  dec=np.array([-1.21788544571165, -1.21302218898355]), 
  frame='icrs', 
  unit='deg'
)
result = SDSS.query_crossid(
  coordinates=pos, 
  photoobj_fields=['objID', 'ra', 'dec'],
  specobj_fields=['specObjID', 'z'], 
  data_release=17,
  radius=2*u.arcsec,
  cache=False
)
```

> Note about DR < 12:
> Queries in data realeases < 12 results in a 404 HTTP status code, before and after this changes.
> Looks like services for old DRs have been removed 

## DR 12 (200 OK)

```http
200 OK http://skyserver.sdss.org/dr12/en/tools/search/X_Results.aspx
Cache-Control: private
Content-Type: text/plain; charset=utf-8
Content-Encoding: gzip
Vary: Accept-Encoding
Server: Microsoft-IIS/8.5
Access-Control-Allow-Origin: *
Content-Disposition: attachment;filename="Skyserver_CrossID1/16/2022 1:11:46 PM.csv"
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Date: Sun, 16 Jan 2022 13:11:45 GMT
Content-Length: 322
```
```
#Table1
obj_id,objID,ra,dec,obj_id1,specObjID,z,type
obj_0,1237663275780276471,359.970818399119,-1.21788545489075,1237663275780276471,435757928612390912,0.1629147,GALAXY
obj_1,1237663275780341888,0.128658777722677,-1.21302220589001,1237663275780341888,435746658618206208,0.07535086,GALAXY
```

## DR 13 (200 OK)

```http
200 OK http://skyserver.sdss.org/dr13/en/tools/search/X_Results.aspx
Cache-Control: private
Content-Type: text/plain; charset=utf-8
Content-Encoding: gzip
Vary: Accept-Encoding
Server: Microsoft-IIS/8.5
Access-Control-Allow-Origin: *
Content-Disposition: attachment;filename="Skyserver_Radial1/16/2022 1:14:47 PM.csv"
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Date: Sun, 16 Jan 2022 13:14:47 GMT
Content-Length: 325
```
```
#Table1
obj_id,objID,ra,dec,obj_id1,specObjID,z,type
obj_0,1237663275780276471,359.970818400315,-1.21788544571165,1237663275780276471,435757928612390912,0.1629147,GALAXY
obj_1,1237663275780341888,0.1286587799259,-1.21302218898355,1237663275780341888,435746658618206208,0.07535086,GALAXY
```

## DR 14 (200 OK)

```http
200 OK http://skyserver.sdss.org/dr14/en/tools/search/X_Results.aspx
Cache-Control: private
Content-Type: text/plain; charset=utf-8
Content-Encoding: gzip
Vary: Accept-Encoding
Server: Microsoft-IIS/8.5
Access-Control-Allow-Origin: *
Content-Disposition: attachment;filename="Skyserver_SQL1/16/2022 1:15:18 PM.csv"
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Date: Sun, 16 Jan 2022 13:15:18 GMT
Content-Length: 325
```
```
#Table1
obj_id,objID,ra,dec,obj_id1,specObjID,z,type
obj_0,1237663275780276471,359.970818400315,-1.21788544571165,1237663275780276471,435757928612390912,0.1629147,GALAXY
obj_1,1237663275780341888,0.1286587799259,-1.21302218898355,1237663275780341888,435746658618206208,0.07535086,GALAXY
```

## DR 15 (200 OK)

```http
200 OK http://skyserver.sdss.org/dr15/en/tools/search/X_Results.aspx
Cache-Control: private
Content-Type: text/plain; charset=utf-8
Content-Encoding: gzip
Vary: Accept-Encoding
Server: Microsoft-IIS/8.5
Access-Control-Allow-Origin: *
Content-Disposition: attachment;filename="Skyserver_CrossID1/16/2022 1:15:42 PM.csv"
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Date: Sun, 16 Jan 2022 13:15:42 GMT
Content-Length: 325
```
```
#Table1
obj_id,objID,ra,dec,obj_id1,specObjID,z,type
obj_0,1237663275780276471,359.970818400315,-1.21788544571165,1237663275780276471,435757928612390912,0.1629147,GALAXY
obj_1,1237663275780341888,0.1286587799259,-1.21302218898355,1237663275780341888,435746658618206208,0.07535086,GALAXY
```

## DR 16 (200 OK)

```http
200 OK http://skyserver.sdss.org/dr16/en/tools/search/X_Results.aspx
Cache-Control: private
Content-Type: text/plain; charset=utf-8
Content-Encoding: gzip
Vary: Accept-Encoding
Server: Microsoft-IIS/8.5
Access-Control-Allow-Origin: *
Content-Disposition: attachment;filename="Skyserver_SQL1/16/2022 1:15:58 PM.csv"
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Date: Sun, 16 Jan 2022 13:15:57 GMT
Content-Length: 325
```
```
#Table1
obj_id,objID,ra,dec,obj_id1,specObjID,z,type
obj_0,1237663275780276471,359.970818400315,-1.21788544571165,1237663275780276471,435757928612390912,0.1629147,GALAXY
obj_1,1237663275780341888,0.1286587799259,-1.21302218898355,1237663275780341888,435746658618206208,0.07535086,GALAXY
```

## DR 17 (200 OK)

```http
200 OK http://skyserver.sdss.org/dr17/en/tools/search/X_Results.aspx
Transfer-Encoding: chunked
Content-Type: text/plain
Content-Encoding: gzip
Vary: Accept-Encoding
Server: Kestrel
Content-Disposition: attachment;filename="Skyserver_CrossID1/16/2022 1:06:29 PM.csv"
X-Powered-By: ASP.NET
Date: Sun, 16 Jan 2022 13:06:29 GMT
```
```
#Table1
obj_id,objID,ra,dec,obj_id1,specObjID,z,type
obj_0,1237663275780276471,359.970818400315,-1.21788544571165,1237663275780276471,435757928612390912,0.1629147,GALAXY
obj_1,1237663275780341888,0.1286587799259,-1.21302218898355,1237663275780341888,435746658618206208,0.07535086,GALAXY
```

### CODE REVIEW

**Review by keflavich — APPROVED** [review]
Nice!  Looks good to me.  Thanks for the thorough writeup.

File: astroquery/sdss/core.py [hunk 1]
```diff
@@ -144,7 +144,10 @@ def query_crossid_async(self, coordinates, obj_names=None,
 
         sql_query += ',dbo.fPhotoTypeN(p.type) as type \
                       FROM #upload u JOIN #x x ON x.up_id = u.up_id \
-                      JOIN PhotoObjAll p ON p.objID = x.objID ORDER BY x.up_id'
+                      JOIN PhotoObjAll p ON p.objID = x.objID '
+        if specobj_fields:
```
Comment (ceb8) [inline astroquery/sdss/core.py:148]: Can you add a test when `specobj_fields` is set? I think that would be very useful to have.

**Review by ceb8 — COMMENTED** [review]
I think this looks great! Would be nice to have an added test for the new option.

File: astroquery/sdss/core.py (hunk 1 above)
Comment (nmcardoso) [reply astroquery/sdss/core.py:148]: I wrote a new use case similar to the following written without spec in commit fb84a65

https://github.com/astropy/astroquery/blob/564406770eaf332f156ca0eba28ee561d041925a/astroquery/sdss/tests/test_sdss_remote.py#L169-L176

**Review by ceb8 — APPROVED** [review]

**Comment by ceb8** [comment]
Thanks @nmcardoso !

**Comment by nmcardoso** [comment]
@ceb8 😀👍

**Comment by bsipocz** [comment]
The test added here were never actually worked. It wasn't picked up by CI by design as remote tests are not run in CI, but they should have been checked before the merge. I'm opening a fix separately.

**Comment by bsipocz** [comment]
Actually, before doing a fix, I wonder how this ever worked for you @nmcardoso? As I actually run into the issue that the raw SQL doesn't work with the error `Invalid column name 'objID`, and the code that generates that part of the SQL doesn't seem to be changing with this PR.

**Comment by bsipocz** [comment]
The fixes and changelog is added for this in #2304

**Pull request merged**